keps/sig-network/2091-cluster-network-policy/README.md

```diff
@@ -0,0 +1,958 @@
+
+```yaml
+apiVersion: networking.k8s.io/v1alpha1
+kind: ClusterNetworkPolicy
+metadata:
+  name: restrict-egress-to-db
+spec:
+  appliedTo:
+    namespaceSelector: {}
+    podSelector:
+      matchExpressions:
+        - {key: app, operator: NotIn, values: [authorized-client]}
+  egress:
+    - action: Deny
+      to:
+      - ipBlock:
+          cidr: 10.220.0.8/32  # restricted database running behind static IP
+```
+
+### Test Plan
+
+- Add e2e tests for ClusterNetworkPolicy resource
+  - Ensure `Deny` rules override all allowed traffic in the cluster
+  - Ensure `Allow` rules override K8s NetworkPolicies
+  - Ensure that in stacked ClusterNetworkPolicies/K8s NetworkPolicies, the following precedence is maintained
+    aggregated `Deny` rules > aggregated `Allow` rules > K8s NetworkPolicy rules
+- Add e2e tests for DefaultNetworkPolicy resource
+  - Ensure that in absence of ClusterNetworkPolicy rules and K8s NetworkPolicy rules, DefaultNetworkPolicy rules are observed
+  - Ensure that K8s NetworkPolicies override DefaultNetworkPolicies by applying policies to the same workloads
+  - Ensure that stacked DefaultNetworkPolicies are additive in nature
+- e2e test cases must cover ingress and egress rules
+- e2e test cases must cover port-ranges, named ports, integer ports etc
+- e2e test cases must cover various combinations of `podSelector` in `appliedTo` and ingress/egress rules
+- e2e test cases must cover various combinations of `namespaceSelector` in `appliedTo`
+- e2e test cases must cover various combinations of `namespaces` in ingress/egress rules
+  - Ensure that `except` field works as expected
+  - Ensure that `self` field works as expected
+- Add unit tests to test the validation logic which shall be introduced for cluster-scoped policy resources
+  - Ensure that `self` field cannot be set along with `selector` within `namespaces`
+  - Test cases for fields which are shared with NetworkPolicy, like `ipBlock`, `endPort` etc.
+- Ensure that only administrators or assigned roles can create/update/delete cluster-scoped policy resources
+
+### Graduation Criteria
+
+#### Alpha to Beta Graduation
+
+- Gather feedback from developers and surveys
+- At least 1 CNI provider must provide the implementation for the complete set
+  of alpha features
+- Evaluate "future work" items based on feedback from community
+
+#### Beta to GA Graduation
+
+- At least 2 CNI providers must provide the implementation for the complete set
+  of alpha features
+- More rigorous forms of testing
+  — e.g., downgrade tests and scalability tests
+- Allowing time for feedback
+- Completion of all accepted "future work" items
+
+#### Removing a Deprecated Flag
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
```

Comment:
There are no flags involved here. This section does not apply; you can just delete it.